jwst/residual_fringe/residual_fringe.py

```diff
@@ -83,11 +83,14 @@ def do_correction(self):
             raise ErrorNoFringeFlat("The fringe flat step has not been run on file %s",
                                     self.input_model.meta.filename)
 
-        # Remove any NaN values from the data prior to processing
+        # Remove any NaN values adn bad pixels from the data prior to processing
         # Set them to 0 for the residual fringe routine
         # They will be re-added at the end
         output_data = self.model.data.copy()
-        nanval_indx = np.where(~np.isfinite(output_data))
+        dq = self.model.dq.copy()
```

Comment:
why does the DQ array need to be copied here?  it doesn't appear that you ever modify it